### PR TITLE
Array binary search

### PR DESCRIPTION
**Pull request by dcalhoun286**

### CODE REVIEW

**Comment by dcalhoun286** [comment]
- [x] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed
 - [ ] Unit tests written and passing
     - [ ] “Happy Path” - Expected outcome
     - [ ] Expected failure
     - [ ] Edge Case (if applicable/obvious)
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [x] Link to code
     - [x] Picture of whiteboard

File: javascript/code-challenges/arrayBinarySearch/array-binary-search.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  while (rangeStart <= rangeEnd) {
+    let mid = Math.floor((rangeStart + rangeEnd)/2);
+
+    if (array[mid] === value) {
+      // array element exists
+      return mid;
+    } else if (array[mid] < value) {
+      rangeEnd = mid;
+    } else if (array[mid] > value) {
+      rangeStart = mid;
```
Comment (lizkavalski) [inline javascript/code-challenges/arrayBinarySearch/array-binary-search.js:17]: switch these two around otherwise your going into an infinite
